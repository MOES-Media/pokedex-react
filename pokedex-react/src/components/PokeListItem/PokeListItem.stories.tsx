import React from 'react';
import { Story, Meta } from '@storybook/react';

import PokeListItem, {PokeListItemProps} from '.';

export default {
    title: "Components/PokeListItem",
    component: PokeListItem
} as Meta;

const Template: Story<PokeListItemProps> = (args) => <PokeListItem {...args} />
export const Default = Template.bind({});
Default.args = {
    sprite: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/back/1.png",
    onClick: () => {console.log("onclick")}
}