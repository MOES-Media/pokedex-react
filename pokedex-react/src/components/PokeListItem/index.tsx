import React, { FC } from "react";
import styled from "styled-components";

interface StyledPokeListItemProps {
  backgroundImage: string;
};

export interface PokeListItemProps {
  sprite: string;
  onClick: (event?: any) => void;
}

const StyledPokeListItem = styled.button<StyledPokeListItemProps>`
  width: 120px;
  height: 120px;
  border: none;
  border-radius: 5px;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.16);
  margin: 10px;
  ${({ backgroundImage }) => `background-image: url(${backgroundImage});`}
  background-position: center;
  background-repeat: no-repeat;
`;

const PokeListItem: FC<PokeListItemProps> = ({ sprite, onClick }) => (
  <StyledPokeListItem backgroundImage={sprite} onClick={onClick} role="button"/>
);

export default PokeListItem;
