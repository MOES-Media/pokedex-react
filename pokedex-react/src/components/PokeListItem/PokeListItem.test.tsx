import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';

import PokeListItem from '.';

// unit testing
test("Component: PokeListItem should call onClick function", () => {
    const mockOnClick = jest.fn(); // mock function 
    
    const pokeListItem = render(<PokeListItem onClick={mockOnClick} sprite="foo-bar"/>); // render in memory
    const pokeListButton = pokeListItem.getByRole("button"); // get button html 
    fireEvent.click(pokeListButton); // fire click event
    
    // assert mock is called and only called once
    expect(mockOnClick).toBeCalled(); 
    expect(mockOnClick).toHaveBeenCalledTimes(1);
});

// snapshot testing
test("Component: PokeListItem Should render correctly", () => {
    const mockOnClick = jest.fn();
    const snapshot = renderer.create(<PokeListItem sprite="http://foo.bar/example.png" onClick={mockOnClick}/>).toJSON();
    expect(snapshot).toMatchSnapshot();
});