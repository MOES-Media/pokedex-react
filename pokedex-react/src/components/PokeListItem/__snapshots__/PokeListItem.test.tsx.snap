// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PokeListItem Should render correctly 1`] = `
<button
  className="sc-bdnxRM hSQkRS"
  onClick={[MockFunction]}
  role="button"
/>
`;
